src/lib/auth.ts

```diff
@@ -5,17 +5,37 @@ import type { User } from "firebase/auth"
 import { app } from './firebase'
 
 export interface AuthState {
+  // auth.user will be either a firebase User object or null
   user: User | null
+  // auth.known is whether or not we have completed trying to login with firebase
+  // --i.e. if `true` then we "know" that we have a user or we "know" that we don't,
+  // but when the store is first initialized we don't "know" anything.
   known: boolean
 }
 
+// TODO: Why is this written as a function returning a store instead of export const auth={...}?
+// (in my testing it seems to work the other way, so I'm guessing just coding style?)
```

Comment:
Just style, from times when some stores are factory methods that take parameters